### PR TITLE
Disable FLoC

### PR DESCRIPTION
**Pull request by kdzwinel**
**Reviewer:** @jonathanKingston 
**Tech Design:** https://app.asana.com/0/481882893211075/1200159597529958/f

## Description:
This disables FLoC API by opting all sites out of FLoC and removing the JS API.

## Steps to test this PR:

1. run chrome (89+) with flags that force FLoC on:
```
/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary --enable-blink-features=InterestCohortAPI --enable-features="FederatedLearningOfCohorts:update_interval/10s/minimum_history_domain_size_required/2,FlocIdSortingLshBasedComputation,InterestCohortFeaturePolicy"
```

1. clear browsing history

1. visit couple of websites

1. verify that document.interestCohort() doesn't return a cohort number (in context where we don't block the API - e.g. safelisted or any chrome:// page)

1. verify that document.interestCohort() is not available (in context where we block the API)

Make sure that:
- this works on sites that alrady have permissions-policy header (e.g. https://scotthelme.co.uk/goodbye-feature-policy-and-hello-permissions-policy/ , https://securityheaders.com/)
- permissions-policy opt-out header should be on all frames (main and sub)
- safelisting the site should bring back the JS API but the opt out header should still be there
- verify that it's not causing issues when running in Chrome with FLoC not force-enabled

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/chrome-events.es6.js [hunk 1]
```diff
@@ -201,26 +219,32 @@ chrome.webRequest.onHeadersReceived.addListener(
...
+        }
+
         if (blockingExperimentActive()) {
-            let responseHeaders = request.responseHeaders
             // Strip 3rd party response header
             const tab = tabManager.get({ tabId: request.tabId })
-            if (!request.responseHeaders) return
-            if (tab && tab.site.whitelisted) return
+            if (!request.responseHeaders) return { responseHeaders: responseHeaders }
+            if (tab && tab.site.whitelisted) return { responseHeaders: responseHeaders }
```
Comment (kdzwinel) [inline shared/js/background/chrome-events.es6.js:233]: that's a bit ugly but I didn't want to completely rewrite that code

File: shared/js/background/chrome-events.es6.js [hunk 2]
```diff
@@ -201,26 +219,32 @@ chrome.webRequest.onHeadersReceived.addListener(
...
+            responseHeaders.push({ name: 'permissions-policy', value: 'interest-cohort=()' })
+        }
+
         if (blockingExperimentActive()) {
-            let responseHeaders = request.responseHeaders
             // Strip 3rd party response header
             const tab = tabManager.get({ tabId: request.tabId })
-            if (!request.responseHeaders) return
-            if (tab && tab.site.whitelisted) return
+            if (!request.responseHeaders) return { responseHeaders: responseHeaders }
```
Comment (jonathanKingston) [inline shared/js/background/chrome-events.es6.js:232]:
```suggestion
            if (!request.responseHeaders) return { responseHeaders }
```

Comment (jonathanKingston) [reply shared/js/background/chrome-events.es6.js:232]: This also suggests sometimes we have no headers, is this on non document like requests? Like `javascript:` perhaps?

Perhaps we have to handle that above too when using responseHeaders.push?

Comment (kdzwinel) [reply shared/js/background/chrome-events.es6.js:232]: Can we safely use `{ responseHeaders }`? I haven't seen that in other parts of the code.

> This also suggests sometimes we have no headers, is this on non document like requests? Like javascript: perhaps?
> Perhaps we have to handle that above too when using responseHeaders.push?

Hm, this is good point, I'll ask Brad - he added that condition.

Comment (kdzwinel) [reply shared/js/background/chrome-events.es6.js:232]: OK we do use the `{ x }` notation in other places 👍

Brad told me that he can't recall in which cases headers were empty, but he remember this happening. I added a safeguard.

**Review by jonathanKingston — APPROVED** [review]
LGTM now! Thanks

File: shared/js/background/chrome-events.es6.js [hunk 3]
```diff
@@ -175,6 +175,24 @@ function blockingExperimentActive () {
...
+
+// Overwrite FLoC JS API
+if (isFlocEnabled) {
+    chrome.webNavigation.onCommitted.addListener(details => {
+        const tab = tabManager.get({ tabId: details.tabId })
+        if (tab && tab.site.whitelisted) return
+
+        chrome.tabs.executeScript(details.tabId, {
+            file: 'public/js/content-scripts/floc.js',
+            allFrames: true,
```
Comment (dharb) [inline shared/js/background/chrome-events.es6.js:189]: See https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/523. Confusingly, `onCommitted` is called when any frame changes location, not just the main frame.

**Review by dharb — CHANGES_REQUESTED** [review]
Looking good! Just one tiny fix, then we should be ready to ship.

File: shared/js/background/chrome-events.es6.js (hunk 3 above)
Comment (kdzwinel) [reply shared/js/background/chrome-events.es6.js]: Thanks Dave, good catch! I addressed it.

**Review by dharb — APPROVED** [review]
👍 💥

**Pull request merged**